p2panda-net/src/sync/mod.rs

```diff
@@ -72,23 +83,28 @@ pub async fn initiate_sync<S: AsyncWrite + Send + Unpin, R: AsyncRead + Send + U
                     header,
                     payload,
                     delivered_from: peer,
-                    topic,
+                    topic: topic_id.into(),
                 })
                 .await
             {
                 error!("error in sync actor: {}", err)
             };
         }
         engine_actor_tx
-            .send(ToEngineActor::SyncDone { peer, topic })
+            .send(ToEngineActor::SyncDone {
+                peer,
+                topic: topic_id.into(),
+            })
             .await
             .expect("engine channel closed");
     });
 
     // Run the sync protocol.
     let result = sync_protocol
         .initiate(
-            topic.as_bytes(),
+            // @TODO: When generic topic is implemented in `p2panda-sync` we can pass the topic
+            // directly here.
```

Comment:
Can this comment be removed? Not sure I understand it, we're passing in the topic and it is generic?